### ISSUE TITLE
Viewer doesn't display factor levels of a data.frame in a list properly

### ISSUE THREAD

**Issue by BartJanvanRossum** [rstudio/rstudio#2195]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.423
    OS Version      :  Windows 7, 64 bit
    R Version       : 3.4.3

### Steps to reproduce the problem

```R
df <- data.frame(v1 = c("b1", "b1"))
ldf <- list(df = df)
View(ldf)
```

### Describe the problem in detail

The data.frame in a list as in the example contains 1 variable, a factor with 1 level. However the viewer in RStudio displays it as a data.frame in a list with 1 variable, a factor with 2 levels.

### Describe the behavior you expected

I expected to see a factor variable with 1 level just it is displayed by str() and also in the environment tab.

I don't think it really matters but this is the sessionInfo:

R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.4.3 tools_3.4.3    yaml_2.1.16

**Comment by dfalty** [comment]
Thanks for the bug report! I was able to reproduce this on 1.1.423 (Mac)

**Comment by kevinushey** [comment]
This should be fixed up on master -- sorry, I've since lost the associated commit where this was fixed, but it will be part of the v1.2 release.
